runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7534,7 +7536,8 @@ static void handleOffHeapDataForArrays(
          // Clear out tempReg if dealing with 0 length array
          zeroReg = srm->findOrCreateScratchRegister();
          generateRegRegInstruction(TR::InstOpCode::XORRegReg(), node, zeroReg, zeroReg, cg);
-         generateRegImmInstruction(TR::InstOpCode::CMPRegImm4(), node, sizeReg, 0, cg);
+         // Since array size is capped at 32 bits, we only need to check 32-63 bits of lengthReg.
+         generateRegImmInstruction(TR::InstOpCode::CMP4RegImm4, node, sizeReg, 0, cg);
```

Comment:
Since you're making changes to this very instruction, a better instruction for 32-bit compares to zero is:
`generateRegRegInstruction(TR::InstOpCode::TEST4RegReg, node, sizeReg, sizeReg, cg);`